pypeit/core/gui/edge_inspector.py

```diff
@@ -0,0 +1,321 @@
+        while len(self.trace_plot) > 0:
+            for i, l in enumerate(self.image_plot.axes.lines):
+                if l is self.trace_plot[0]:
+                    break
+            self.image_plot.axes.lines[i].remove()
+            del self.trace_plot[0]
+
+        # Reset the traces using the updated edges object
+        self._reset_traces()
+        # Plot the traces
+        self.plot_traces()
+        # Update the plot
+        pyplot.draw()
+            
+    def _reset_traces(self):
+        """
+        Re-initialize the internals that perform the trace bookkeeping.
+        """
+        self.spec_pix = np.arange(self.edges.nspec)
+        self.ntrace = self.edges.ntrace
+        self.good = self.edges.good_traces(include_box=True)
+        self.side = np.clip(self.edges.traceid, -1, 1) 
+        self.add = np.zeros(self.ntrace, dtype=bool)
+        self.remove = np.zeros(self.ntrace, dtype=bool)
+        self.offset = np.zeros(self.ntrace, dtype=float)
+        self.trace_cen = self.edges.edge_cen if self.edges.edge_fit is None \
+                            else self.edges.edge_fit
+        
+    def undo(self, pos):
+        """
+        Undo all operations since the last time the ``edges`` object was
+        updated.
+        """
+        self._reset_traces()
+        self.update_traces()
+        
+    def nearest_trace(self, spat):
+        """
+        Find the trace nearest to the spatial position of the cursor.
+
+        Args:
+            spat (:obj:`float`):
+                The spatial (x) position of the cursor
+        """
+        # Only select from the traces that are good and not being removed
+        indx = self.good & np.logical_not(self.remove)
+        diff = np.absolute(self.trace_cen[self.reference_row][indx] + self.offset[indx] - spat)
+        return np.arange(self.ntrace)[indx][np.argmin(diff)]
+        
+    def move(self, pos):
+        """
+        Move the trace nearest the cursor spatial (x) position to the cursor's
+        location.
+
+        Args:
+            pos (:obj:`list`):
+                List with x and y position of the cursor at the time of the
+                window event.
+        """
+        # Find the nearest trace
+        i = self.nearest_trace(pos[0])
+        # Get the spatial offset
+        self.offset[i] = pos[0] - self.trace_cen[self.reference_row,i]
+        # Report
+        msgs.info(f'Offsetting trace {i} by {self.offset[i]} pixels')
+        # Reset the line data
+        self.trace_plot[i].set_data((self.trace_cen[:,i] + self.offset[i], self.spec_pix))
+        # Update the plot
+        pyplot.draw()
+
+    def delete(self, pos):
+        """
+        Delete the trace nearest the cursor spatial (x) position.
+
+        Args:
+            pos (:obj:`list`):
+                List with x and y position of the cursor at the time of the
+                window event.
+        """
+        # Find the nearest trace
+        i = self.nearest_trace(pos[0])
```

Comment:
could we just pass in the zeroth element of `pos`, since pos[1] is never used?

Reply:
Good question!  The issue is that all the functions connected to pointer events have to have the same calling sequence, which is set by `_set_event` in `gui_util.Pointer`.  I added a few doc-string sentences to clarify this.